### ISSUE TITLE
Error in pipeline after updating to the latest version

### ISSUE THREAD

**Issue by faritvidalGL** [awslabs/aws-service-catalog-puppet#622]
Please include a link to your expanded manifest, the full contents of your AWS CodeBuild output (see https://aws-service-catalog-puppet.readthedocs.io/en/latest/puppet/using_the_cli.html#export-puppet-pipeline-logs)

Please ensure you are using the latest version and have run a validate command on your manifest file see (https://aws-service-catalog-puppet.readthedocs.io/en/latest/puppet/using_the_cli.html#validate)

# Steps to reproduce
1. Update from version 0.143.0 to 0.213.1
1. run the pipeline named servicecatalog-puppet-pipeline
 
# Expected results
Error **"Exception: invalid use of launch affinity - product-1 is not deployed for task launches_product-2_************-us-east-1"**

Example of manifest snippet:

> product-1:
>     portfolio: internal-portfolio
>     product: product-1
>     version: *product-1-version
>     deploy_to:
>       tags:
>         - regions: default_region
>            tag: app:product-1
> 
>   product-2:
>     depends_on:
>       - product-1
>     portfolio: internal-portfolio
>     product: product-2
>     version: *product-2-version
>     deploy_to:
>       tags:
>         - regions: enabled_regions
>            tag: app:product-2

the pipeline fails on every product that has a depends_on assigned

# Actual results

[log-events-viewer-result.csv](https://github.com/awslabs/aws-service-catalog-puppet/files/10395064/log-events-viewer-result.csv)

**Comment by eamonnfaherty** [comment]
Can I see your expanded manifest file please.

**Comment by faritvidalGL** [comment]
It is possible to send you the manifest file in a more direct way without issue attachment, to avoid posting account IDs and other data, maybe email, stay tuned

**Comment by puddleglum1904** [comment]
@eamonnfaherty expanded manifest has been sent via email.

**Comment by puddleglum1904** [comment]
Still receiving this error in version 0.232.0 when trying to deploy a product that depends on another product in another account (all part of the same manifest file). This error occurs as part of the command `Running command servicecatalog-puppet --info generate-task-reference $PWD/manifest-expanded.yaml`
i.e.
PRODUCT-A is configured to be deployed in ACCOUNT-A
PRODUCT-B depends on PRODUCT-A, and is configured to be deployed in ACCOUNT-B

Error:
```
Exception: invalid use of launch affinity - {PRODUCT-A} is not deployed for task launches_{PRODUCT-B}_{ACCOUNT-B}_us_east_1
```

@eamonnfaherty I've emailed the manifest file to you, this error only appears to occur when the dependency is in a different account

**Comment by eamonnfaherty** [comment]
this is resolved  in [0.234.0](https://github.com/awslabs/aws-service-catalog-puppet/releases/tag/0.234.0)
